### PR TITLE
Allow multiple network selection

### PR DESCRIPTION
**Pull request by Pavneet-Sing**
## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

[multi-network-selector.webm](https://github.com/brave/brave-core/assets/11755381/b43e43e2-8595-4d7c-82e7-2e2ab3e34179)

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/app/domain/PortfolioModel.java [hunk 1]
```diff
@@ -150,28 +152,52 @@ public void fetchAssetsByType(TokenUtils.TokenType type, NetworkInfo selectedNet
...
+     * Fetch the assets of selected networks and filter by type.
+     * @param type by NFT, NON-NFT, ERC, SOL etc.
+     * @param selectedNetworks list to fetch assets from
+     * @param braveWalletBaseActivity instance for native services
+     * @param callback to get the @{code PortfolioHelper} object containing result
+     */
+    public void fetchAssets(TokenUtils.TokenType type, List<NetworkInfo> selectedNetworks,
+            BraveWalletBaseActivity braveWalletBaseActivity,
+            Callbacks.Callback1<PortfolioHelper> callback) {
+        synchronized (mLock) {
```
Comment (Pavneet-Sing) [inline android/java/org/chromium/chrome/browser/app/domain/PortfolioModel.java:187]: I believe we don't need `synchronized`. LMK if I should remove this `synchronized` block.

File: android/java/res/drawable/ic_menu_close.xml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (simoarpe) [inline android/java/res/drawable/ic_menu_close.xml:1]: Missing header.

File: android/java/org/chromium/chrome/browser/app/domain/PortfolioModel.java (hunk 1 above)
Comment (simoarpe) [reply android/java/org/chromium/chrome/browser/app/domain/PortfolioModel.java:187]: `fetchAssets` is always called from UI thread, so I feel that there's no need to make it `synchronized`. We can simply add `MainThread` annotation. And if we want to make it even more strict, include `ThreadUtils.assertOnUiThread()`.

Comment (Pavneet-Sing) [reply android/java/org/chromium/chrome/browser/app/domain/PortfolioModel.java:187]: `ThreadUtils.assertOnUiThread()` is required for UI specific work so not needed here. Will remove it as suspected.

File: android/java/org/chromium/chrome/browser/app/domain/NetworkSelectorModel.java [hunk 3]
```diff
@@ -121,10 +135,17 @@ public LiveData<NetworkInfo> getSelectedNetwork() {
...
 
+    public void setSelectedNetworks(List<NetworkInfo> networkInfos) {
+        _mSelectedNetworks.postValue(networkInfos);
+    }
+
     public Mode getMode() {
         return mMode;
     }
 
+    public SelectionMode getmSelectionType() {
```
[the remark this answers is not in the document]
Comment (Pavneet-Sing) [reply android/java/org/chromium/chrome/browser/app/domain/NetworkSelectorModel.java]: Updated 6f5d135

**Review by mkarolin — APPROVED** [review]
`strings`++

**Pull request merged**